nix/overlay.nix

```diff
@@ -0,0 +1,78 @@
+tensorflowSubModule: final: prev:
+
+let
+  /* The tensorflow submodule in third_party/tensorflow and the symlinks in the
+     Haskell-package directories will always be removed by Nix.
+     Thus construct a source here where the submodule is a normal directory
+     whenever required.
+  */
+  tfSrc = prev.runCommand "prepTensorflowSrc" {
+    src = prev.nix-gitignore.gitignoreSource [
+      "nix/"
+      "flake.nix"
+      "flake.lock"
+      "result"
+    ] ./..;
+  } ''
+    cp -r $src/* .
+
+    chmod -R +rwx ./third_party
+    cp -r ${tensorflowSubModule} ./third_party/tensorflow
```

Comment:
While I don't see a good option to get Nix accepting the submodules, it would be nice to use `prev.libtensorflow.src`  instead `tensorflowSubModule`, so that the correct version also used in nixpkgs is provided. libtensorflow-2.9.1 from current nixpkgs gives compilation errors, however.

Reply:
I have no opinion about this as I don't understand Nix enough and we have no CI around that. Fix it to your liking?